misc/services/camel/de-metas-camel-externalsystems/de-metas-camel-grssignum/src/main/java/de/metas/camel/externalsystems/grssignum/from_grs/product/processor/PushRawMaterialsProcessor.java

```diff
@@ -125,4 +126,20 @@ private static String getName(@NonNull final JsonProduct grsProduct)
 
 		return name;
 	}
+
+	@NonNull
+	private static String computeBPartnerIdentifier(@NonNull final JsonBPartnerProduct jsonBPartnerProduct)
+	{
+		if (jsonBPartnerProduct.getBPartnerMetasfreshId() != null && Check.isNotBlank(jsonBPartnerProduct.getBPartnerMetasfreshId()))
+		{
+			return jsonBPartnerProduct.getBPartnerMetasfreshId();
+		}
+
+		if (jsonBPartnerProduct.getBpartnerId() != null && Check.isNotBlank(jsonBPartnerProduct.getBpartnerId()))
+		{
+			return ExternalIdentifierFormat.asExternalIdentifier(jsonBPartnerProduct.getBpartnerId());
+		}
+
```

Comment:
```suggestion
```

i think it's better to use the metasfresh-id (as aggreed with GRS) or fail. trying to fall back will probably also fail, but with more FUD